src/test/rgw/rgw_multi/tests.py

```diff
@@ -957,9 +963,22 @@ def test_datalog_autotrim():
 
     # trim each datalog
     for zone, _ in zone_bucket:
+        # read max markers for each shard
+        status = datalog_status(zone.zone)
+
         datalog_autotrim(zone.zone)
-        datalog = datalog_list(zone.zone)
-        assert len(datalog) == 0
+
+        for shard_id, shard_status in enumerate(status):
+            try:
+                before_trim = dateutil.parser.isoparse(shard_status['last_update'])
+            except: # empty timestamps look like "0.000000" and will fail here
+                continue
```

Comment:
on exception you just move to the next shard, nothing fails in the test

Reply:
that's the intent, yeah. if there are no entries in the log shard, its 'last_update' timestamp will be "0.000000" and we hit this exception. because the shard was empty, we can skip the listing and timestamp comparison